src/JustSaying/AwsTools/MessageHandling/SnsTopicByName.cs

```diff
@@ -85,49 +85,50 @@ public override async Task<bool> ExistsAsync()
             return false;
         }
 
-        public async Task<bool> CreateAsync()
+        public async Task CreateAsync()
         {
             try
             {
                 var response = await Client.CreateTopicAsync(new CreateTopicRequest(TopicName))
                     .ConfigureAwait(false);
 
-                if (!string.IsNullOrEmpty(response.TopicArn))
+                if (string.IsNullOrEmpty(response.TopicArn))
                 {
-                    Arn = response.TopicArn;
-                    _log.LogInformation("Created topic '{TopicName}' on ARN '{Arn}'.", TopicName, Arn);
-                    return true;
+                    _log.LogError("Failed to create or obtain ARN for topic {TopicName}.", TopicName);
+                    throw new Exception($"Failed to create or obtain ARN for topic '{TopicName}'.");
```

Comment:
Let's use `InvalidOperationException` instead of `Exception`.

Reply:
Is there more context that can be added to the message here? Like something about the ARN not being retrieved?

Reply:
Should we be throwing an InvalidOperationException for something the developer can't fix?

We can add the request id that comes back, but there's no error message or anything aside from a map of repsonse metadata. 

The message could be changed to be something closer to retrieval failure?

Reply:
Have changed this a bit, let me know what you think.